package/src/models/forms/universal.ts

```diff
@@ -520,6 +525,11 @@ const universalMachineConfig: AuthMachineConfig = {
         "setupView",
       ],
       always: [
+        // Handle no first factors
+        {
+          target: "noFirstFactors",
```

Comment:
In theory this should be after the following `target: handleLoginWithLink, cond: hasLinkQueryParams` transition. 

In practice I don't think it matters, because I think it only changes the outcome in a very silly corner case:

1. User loads login page, first factors are present
2. User successfully uses SMS verification code as their first factor
3. User selects email link as their second factor
4. Site admin disables all first factors, but leaves second factors on (I'm not sure if the dashboard even allows this, but I think it's theoretically possible via API)
5. User clicks the email link and returns to the login page

Expected result: login should succeed, user should be redirected or see Success view

Actual result: user sees "login disabled"

In the case when first factors are disabled while the user is trying to log in with an email link first factor, I think either outcome is fine:

1. User loads login page, first factors are present
2. User selects email link as their first factor
3. Site admin disables all first factors
4. User clicks the email link and returns to the login page

Current result: we try to log in with link, get an error message back from the server, and display the error message

New result: user sees "login disabled"